File: app/src/main/java/com/antyzero/smoksmog/error/SnackBarErrorReporter.java
package com.antyzero.smoksmog.error;

import android.app.Activity;
import android.support.annotation.StringRes;
import android.support.design.widget.Snackbar;

/**
 * Error reporting via Snackbar
 */
public class SnackBarErrorReporter implements ErrorReporter {

    private static final int DURATION = Snackbar.LENGTH_LONG;

    private final Activity activity;

    public SnackBarErrorReporter( Activity activity ) {
        this.activity = activity;
    }

    @Override
    public void report( String message ) {
        processSnackbar( Snackbar.make( activity.findViewById( android.R.id.content ), message, DURATION ) );
    }

    @Override
    public void report( @StringRes int stringId ) {
        processSnackbar( Snackbar.make( activity.findViewById( android.R.id.content ), stringId, DURATION ) );
    }

    @Override
    public void report( @StringRes int stringId, Object... objects ) {
        String message = activity.getResources().getString( stringId, objects );
        processSnackbar( Snackbar.make( activity.findViewById( android.R.id.content ), message, DURATION ) );
    }

    private void processSnackbar( Snackbar snackbar ) {
        snackbar.show();
    }
}

File: app/src/main/kotlin/com/antyzero/smoksmog/error/SnackBarErrorReporter.kt
package com.antyzero.smoksmog.error

import android.app.Activity
import android.support.annotation.StringRes
import android.support.design.widget.Snackbar

/**
 * Error reporting via SnackBar
 */
class SnackBarErrorReporter(private val activity: Activity) : ErrorReporter {

    override fun report(message: String) {
        processSnackBar(Snackbar.make(activity.findViewById(android.R.id.content), message, DURATION))
    }

    override fun report(@StringRes stringId: Int) {
        processSnackBar(Snackbar.make(activity.findViewById(android.R.id.content), stringId, DURATION))
    }

    override fun report(@StringRes stringId: Int, vararg objects: Any) {
        val message = activity.resources.getString(stringId, *objects)
        processSnackBar(Snackbar.make(activity.findViewById(android.R.id.content), message, DURATION))
    }

    private fun processSnackBar(snackBar: Snackbar) {
        snackBar.show()
    }

    companion object {
        private val DURATION = Snackbar.LENGTH_LONG
    }
}
